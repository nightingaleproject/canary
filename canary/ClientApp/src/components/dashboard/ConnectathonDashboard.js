import React, { Component } from 'react';
import { Container, Divider, Grid, Header, Icon, Item } from 'semantic-ui-react';
import { DashboardItem } from './DashboardItem';

export class ConnectathonDashboard extends Component {
  displayName = ConnectathonDashboard.name;

  getUrl(id) {
    if(this.props.match.params.type === "message")
    {
      return `/test-connectathon-messaging/${id}`
    }
    else
    {
      return `/test-connectathon/${id}`
    }
  }

  render() {
    return (
      <React.Fragment>
        <Grid centered columns={1}>
          <Grid.Column width={15}>
            <Container className="p-b-50">
              <Divider horizontal>
                <Header as="h2">
                  <Icon name="clipboard list" />
                  IHE Connectathon 2020 {this.props.match.params.type} testing
                </Header>
              </Divider>
              <Item.Group className="m-h-30">
                <DashboardItem
                  icon="male"
                  title="Cancer"
                  description="#000050; Janet Page; Congestive Heart Failure"
                  route={ this.getUrl(1) }
                />
                <DashboardItem
                  icon="male"
                  title="Opioid Death at Home"
                  description="#000051; Madelyn Patel; Cocaine toxicity"
                  route={ this.getUrl(2) }
                />
                <DashboardItem
                  icon="male"
                  title="Pregnant"
                  description="#000052; Vivienne Lee Wright; Cardiopulmonary arrest"
                  route={ this.getUrl(3) }
                />
                <DashboardItem
                  icon="male"
                  title="Car accident at work: Full"
                  description="#000053; Javier Luis Perez; Blunt head trama"
                  route={ this.getUrl(4) }
                />
                <DashboardItem
                  icon="male"
                  title="Car accident at work: Partial"
                  description="#000054; Javier Luis Perez; Blunt head trama (partial record)"
                  route={ this.getUrl(5) }
                />
              </Item.Group>
            </Container>
          </Grid.Column>
        </Grid>
      </React.Fragment>
    );
  }
}
