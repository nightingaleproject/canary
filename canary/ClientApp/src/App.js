import React, { Component } from 'react';
import { Routes, Route, useParams } from 'react-router-dom';
import { ConnectathonDashboard } from './components/dashboard/ConnectathonDashboard';
import { Dashboard } from './components/dashboard/Dashboard';
import { Layout } from './components/Layout';
import { Connectathon } from './components/tests/Connectathon';
import { EDRSRoundtripConsuming } from './components/tests/EDRSRoundtripConsuming';
import { EDRSRoundtripProducing } from './components/tests/EDRSRoundtripProducing';
import { FHIRIJEValidatorProducing } from './components/tests/FHIRIJEValidatorProducing';
import { FHIRConsuming } from './components/tests/FHIRConsuming';
import { FHIRMessageProducing } from './components/tests/FHIRMessageProducing';
import { FHIRProducing } from './components/tests/FHIRProducing';
import { RecentTests } from './components/tests/RecentTests';
import { FHIRCreator } from './components/tools/FHIRCreator';
import { FHIRInspector } from './components/tools/FHIRInspector';
import { FHIRMessageCreator } from './components/tools/FHIRMessageCreator';
import { FHIRMessageSyntaxChecker } from './components/tools/FHIRMessageSyntaxChecker';
import { FHIRSyntaxChecker } from './components/tools/FHIRSyntaxChecker';
import { IJEInspector } from './components/tools/IJEInspector';
import { MessageConnectathonProducing } from './components/tests/MessageConnectathonProducing';
import { RecordConverter } from './components/tools/RecordConverter';
import { RecordGenerator } from './components/tools/RecordGenerator';

export default class App extends Component {
  displayName = App.name;

  render() {
    const FHIRProducingParams = addParams(FHIRProducing)
    const FHIRConsumingParams = addParams(FHIRConsuming)
    const FHIRMessageProducingParams = addParams(FHIRMessageProducing)
    const FHIRMessageCreatorParams = addParams(FHIRMessageCreator)
    const EDRSRoundtripConsumingParams = addParams(EDRSRoundtripConsuming)
    const EDRSRoundtripProducingParams = addParams(EDRSRoundtripProducing)
    const ConnectathonDashboardParams = addParams(ConnectathonDashboard)
    const ConnectathonParams = addParams(Connectathon)
    const MessageConnectathonProducingParams = addParams(MessageConnectathonProducing)

    return (
      <Layout>
        <Routes>
          <Route exact path="/" element={<Dashboard />} />
          <Route path="recent-tests" element={<RecentTests />} />
          <Route path="test-fhir-consuming">
            <Route index element={<FHIRConsumingParams />} />
            <Route path=":id" element={<FHIRConsumingParams />} />
          </Route>
          <Route path="test-fhir-producing">
            <Route index element={<FHIRProducingParams />} />
            <Route path=":id" element={<FHIRProducingParams />} />
          </Route>
          <Route path="test-fhir-message-producing">
            <Route index element={<FHIRMessageProducingParams />} />
            <Route path=":id" element={<FHIRMessageProducingParams />} />
          </Route>
          <Route path="test-fhir-message-creation">
            <Route index element={<FHIRMessageCreatorParams />} />
            <Route path=":id" element={<FHIRMessageCreatorParams />} />
          </Route>
          <Route path="test-edrs-roundtrip-consuming">
            <Route index element={<EDRSRoundtripConsumingParams />} />
            <Route path=":id" element={<EDRSRoundtripConsumingParams />} />
          </Route>
          <Route path="test-edrs-roundtrip-producing">
            <Route index element={<EDRSRoundtripProducingParams />} />
            <Route path=":id" element={<EDRSRoundtripProducingParams />} />
          </Route>
          <Route path="test-fhir-ije-validator-producing" element={<FHIRIJEValidatorProducing />} />
          <Route path="test-connectathon-dash/:type" element={<ConnectathonDashboardParams />} />
          <Route path="test-connectathon/:id" element={<ConnectathonParams />} />
          <Route path="test-connectathon-messaging/:id" element={<MessageConnectathonProducingParams />} />
          <Route path="tool-fhir-inspector" element={<FHIRInspector />} />
          <Route path="tool-fhir-creator" element={<FHIRCreator />} />
          <Route path="tool-fhir-syntax-checker" element={<FHIRSyntaxChecker />} />
          <Route path="tool-fhir-message-syntax-checker" element={<FHIRMessageSyntaxChecker />} />
          <Route path="tool-ije-inspector" element={<IJEInspector />} />
          <Route path="tool-record-converter" element={<RecordConverter />} />
          <Route path="tool-record-generator" element={<RecordGenerator />} />
        </Routes>
      </Layout>
    );
  }
}

/**
 * Adds parameters to the given component class type.
 * @param {*} WrappedComponent The component type to add prarams to.
 * @returns A new version of the component type with parameters.
 */
const addParams = WrappedComponent => () => {
  const params = useParams();
  return (
    <WrappedComponent
      params={params}
    />
  );
};