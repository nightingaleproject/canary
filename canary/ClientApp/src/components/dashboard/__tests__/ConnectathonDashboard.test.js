import renderer from 'react-test-renderer';
import {
    MemoryRouter,
    BrowserRouter,
    Routes, Route, useParams
} from 'react-router-dom';
import { ConnectathonDashboard } from '../ConnectathonDashboard';
import {addParams} from '../../../App'


jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
        type: 'message',
    }),
}));

it('renders', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <ConnectathonDashboard params={{ type: "message" }} />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});