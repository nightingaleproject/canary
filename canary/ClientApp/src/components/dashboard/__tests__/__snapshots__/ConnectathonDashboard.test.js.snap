// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="ui centered one column grid"
>
  <div
    className="fifteen wide column"
  >
    <div
      className="ui container p-b-50"
    >
      <div
        className="ui horizontal divider"
      >
        <h2
          className="ui header"
        >
          <i
            aria-hidden="true"
            className="clipboard list icon"
            onClick={[Function]}
          />
          Connectathon 
          message
           testing
        </h2>
      </div>
      <div
        className="ui items m-h-30"
      />
    </div>
  </div>
</div>
`;
